samcli/cli/context.py

```diff
@@ -139,4 +142,7 @@ def _refresh_session(self):
         the Boto3's session object are read-only. Therefore when Click parses new AWS session related properties (like
         region & profile), it will call this method to create a new session with latest values for these properties.
         """
-        boto3.setup_default_session(region_name=self._aws_region, profile_name=self._aws_profile)
+        try:
+            boto3.setup_default_session(region_name=self._aws_region, profile_name=self._aws_profile)
+        except botocore.exceptions.ProfileNotFound as ex:
+            raise UserException(str(ex))
```

Comment:
Should we have more explicit exceptions? All we will see in Telemetry is UserException is thrown (which happens in a ton of places).

Reply:
Sure, can add an explicit exception.